virttest/qemu_storage.py

```diff
@@ -151,6 +152,32 @@ def get_image_repr(image, params, root_dir, representation=None):
     return func(image, params, root_dir)
 
 
+def create_image_offline_snapshot(params, vm_name, image_name):
+    """
+    Create offline snapshot for image.
+
+    :param params: The image parameters.
+    :type params: dict
+    :param vm_name: The name of the corresponding VM.
+    :type vm_name: str
+    :param image_name: The image name.
+    :type image_name: str
+    :return: The filename of the offline snapshot image.
+    :rtype: str
+    """
+    base_img_filename = storage.get_image_filename(params,
+                                                   data_dir.get_data_dir())
```

Comment:
Instead of using `data_dir.get_data_dir()`, the argument should be given by the caller via `data_root`.

Reply:
Thanks, updated.